coding.md

```diff
@@ -107,12 +129,33 @@ Local variables should be defined before the function code according to ANSI C 8
     }
 ```
 
+## Local variables - libphoenix, userspace
+
+Scope of local variables should be minimalized, as the stack usage and number of local variables. It is advised to avoid reusing variables for different purposes across the function. Static local variables are allowed.
+
+```c
+    void countSheeps(herd_t *herd)
+    {
+        static int lastCount = 0;
+        int count = 0;
+
+        for (size_t i = 0; i < sizeof(herd->sheeps) / sizeof(herd->sheeps[0]); ++i) {
+            if (herd->sheeps[i] != NULL) {
+                ++count;
+            }
+        }
+
+        printf("Counted %d sheeps (last time it was %d)\n", count, lastCount);
+        lastCount = count;
+    }
+```
+
 ## Global variables
 
-Global variables should be used only if they're absolutely necessary. You should avoid using globally initialized variables. If they are used, global variables can only be placed in common structures. The structure should be named after the system module that implements it, followed by _common. Example notation is shown below.
+In the kernel code global variables should be always initialized in runtime. Global variables should be used only if they're absolutely necessary. Scope should be limited whenever possible by using `static`. If they are used, global variables can only be placed in common structures. The structure should be named after the system module that implements it, followed by _common. Example notation is shown below.
```

Comment:
One more thing that  I didn't spot before. I don't really like this rule with obligatory prefixing common struct  with module name. As long as this is static variable it shouldn't be needed. 
Prefixing with module name causes that using this variable is really long sometimes:
`if ((priority >= 0) && (priority >= sizeof(threads_common.ready) / sizeof(threads_common.ready[0]))) {`
vs
`if ((priority >= 0) && (priority >= sizeof(common.ready) / sizeof(common.ready[0]))) {`


Only pros for that is to minimalize issues when someone forgot to add static. On the other hand then it can be easier detected - when someone else forgot to add it again then it should provoke compilation error.

I know that much of our code uses module_common, but please consider it.

Reply:
Exemption added